pkg/server/deploy/handlers.go

```diff
@@ -0,0 +1,100 @@
+package deploy
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"github.com/luizalabs/teresa-api/models/storage"
+	dpb "github.com/luizalabs/teresa-api/pkg/protobuf/deploy"
+)
+
+const (
+	keepAliveMessage = "\u200B" // Zero width space
+)
+
+type Options struct {
+	KeepAliveTimeout     time.Duration `split_words:"true" default:"30s"`
+	RevisionHistoryLimit int           `split_words:"true" default:"5"`
+}
+
+type Service struct {
+	ops     Operations
+	options *Options
+}
+
+func (s *Service) Make(stream dpb.Deploy_MakeServer) error {
+	var appName, description string
+	content := new(bytes.Buffer)
```

Comment:
I disagree with a temporary file because we need to "untar" content, look for teresa.yaml file and send to s3 (as an `io.ReadSeeker`), in other words, if we put the content in a temporary file and erase buffer, we'll need to read it again (in memory I suppose).

I know, these tar files are big (maybe huge). I didn't find a good way to deal with their though.